test/e2e_tests/namespace.go

```diff
@@ -0,0 +1,182 @@
+			Expect(expectedResponse[testutils.CountKey]).To(BeNumerically("<=", response[testutils.CountKey]))
+			for _, ns := range expectedResponse[testutils.NamespaceKey].([]map[string]interface{}) {
+				Expect(ns).To(BeElementOf(response[testutils.NamespaceKey]))
+			}
+		})
+
+		It("Should not get namespaces without a Managed label", func() {
+			baseURI := fmt.Sprintf("%s/v1/%s", platformURL, testutils.NamespaceKey)
+			status, response := performAuthorizedHTTPRequest(httpClient, nil, http.MethodGet, baseURI, "", "", userToken)
+
+			Expect(status).Should(Equal(http.StatusOK))
+			Expect(testutils.KubeSystem).ToNot(BeElementOf(response[testutils.NamespaceKey]))
+		})
+	})
+
+	Context("Validate get Namespace route", func() {
+		It("Should get a specific namespace", func() {
+			baseURI := fmt.Sprintf("%s/v1/%s/%s", platformURL, testutils.NamespaceKey, oneNamespaceName)
+			status, response := performAuthorizedHTTPRequest(httpClient, nil, http.MethodGet, baseURI, "", "", userToken)
+
+			expectedResponse := map[string]interface{}{
+				testutils.NameKey: oneNamespaceName,
+			}
+
+			Expect(status).Should(Equal(http.StatusOK))
+			compareResponses(response, expectedResponse)
+		})
+
+		It("Should handle getting a not found namespace", func() {
+			baseURI := fmt.Sprintf("%s/v1/%s/%s", platformURL, testutils.NamespaceKey, oneNamespaceName+testutils.NonExistentSuffix)
+			status, response := performAuthorizedHTTPRequest(httpClient, nil, http.MethodGet, baseURI, "", "", userToken)
+
+			expectedResponse := map[string]interface{}{
+				testutils.DetailsKey: fmt.Sprintf("%s %q not found", testutils.NamespaceKey, oneNamespaceName+testutils.NonExistentSuffix),
+				testutils.ErrorKey:   testutils.OperationFailed,
+			}
+
+			ns := mocks.PrepareNamespace(oneNamespaceName+testutils.NonExistentSuffix, map[string]string{})
+			Expect(doesResourceExist(k8sClient, &ns)).To(BeFalse())
+			Expect(status).Should(Equal(http.StatusNotFound))
+			compareResponses(response, expectedResponse)
+		})
+	})
+
+	Context("Validate create Namespace route", func() {
+		It("Should create a new namespace", func() {
+			newNamespaceName := generateName(e2eNamespace)
+
+			baseURI := fmt.Sprintf("%s/v1/%s", platformURL, testutils.NamespaceKey)
+			requestData := mocks.PrepareNamespaceType(newNamespaceName)
+			payload, err := json.Marshal(requestData)
+			Expect(err).Should(Not(HaveOccurred()))
+
+			status, response := performAuthorizedHTTPRequest(httpClient, bytes.NewBuffer(payload), http.MethodPost, baseURI, "", "", userToken)
+			expectedResponse := map[string]interface{}{
+				testutils.NameKey: newNamespaceName,
+			}
+
+			Expect(status).Should(Equal(http.StatusOK))
+			compareResponses(response, expectedResponse)
+
+			ns := mocks.PrepareNamespace(newNamespaceName, map[string]string{})
+			Eventually(func() bool {
+				return doesResourceExist(k8sClient, &ns)
+			}, testutils.Timeout, testutils.Interval).Should(BeTrue())
+
+			deleteResource(k8sClient, &ns)
```

Comment:
why it is importent to delete this resource

Reply:
Since we don't allow creation of namespaces with labels, this namespace is not created with the `e2e` label, which is later used to clean up all the `e2e` namespaces. I add add a label to the namespace instead of deletion it here, and it will just be cleaned up later

Reply:
maybe we can add issue to allow create namespaces with label

Reply:
I don't know if that's something we're going to need